cmd/helm/list.go

```diff
@@ -233,23 +246,98 @@ func (l *listCmd) statusCodes() []release.Status_Code {
 	return status
 }
 
-func formatList(rels []*release.Release, colWidth uint) string {
-	table := uitable.New()
-
-	table.MaxColWidth = colWidth
-	table.AddRow("NAME", "REVISION", "UPDATED", "STATUS", "CHART", "APP VERSION", "NAMESPACE")
+func getListResult(rels []*release.Release, next string) listResult {
+	listReleases := []listRelease{}
 	for _, r := range rels {
 		md := r.GetChart().GetMetadata()
-		c := fmt.Sprintf("%s-%s", md.GetName(), md.GetVersion())
 		t := "-"
 		if tspb := r.GetInfo().GetLastDeployed(); tspb != nil {
 			t = timeconv.String(tspb)
 		}
-		s := r.GetInfo().GetStatus().GetCode().String()
-		v := r.GetVersion()
-		a := md.GetAppVersion()
-		n := r.GetNamespace()
-		table.AddRow(r.GetName(), v, t, s, c, a, n)
+
+		lr := listRelease{
+			Name:       r.GetName(),
+			Revision:   r.GetVersion(),
+			Updated:    t,
+			Status:     r.GetInfo().GetStatus().GetCode().String(),
+			Chart:      fmt.Sprintf("%s-%s", md.GetName(), md.GetVersion()),
+			AppVersion: md.GetAppVersion(),
+			Namespace:  r.GetNamespace(),
+		}
+		listReleases = append(listReleases, lr)
+	}
+
+	return listResult{
+		Releases: listReleases,
+		Next:     next,
 	}
-	return table.String()
+}
+
+func shortenListResult(result listResult) []string {
+	names := []string{}
+	for _, r := range result.Releases {
+		names = append(names, r.Name)
+	}
+
+	return names
+}
+
+func formatResult(format string, short bool, result listResult, colWidth uint) (string, error) {
+	var output string
+	var err error
+
+	var shortResult []string
+	var finalResult interface{}
+	if short {
+		shortResult = shortenListResult(result)
+		finalResult = shortResult
+	} else {
+		finalResult = result
+	}
+
+	switch format {
+	case "":
```

Comment:
We should probably accept `--format=plain` as an available option instead, and make that the default. `--format=""` seems a little odd from a UX perspective on its intended behaviour. What do you think?

Reply:
Actually it looks like this is consistent with `helm status --output=""`, so we already have precedent here.